controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/ApplicationApiHandler.java

```diff
@@ -2718,5 +2723,21 @@ private static boolean isOperator(HttpRequest request) {
                               .anyMatch(definition -> definition == RoleDefinition.hostedOperator);
     }
 
+    private void ensureApplicationExistsForPublic(TenantAndApplicationId id, HttpRequest request) {
+        if (controller.system().isPublic() && controller.applications().getApplication(id).isEmpty()) {
```

Comment:
I think we can enable this for non-public as well. We do not send any body when creating an application control and `AthenzAccessControlRequests::credentials` only checks the body if a domain is not set on tenant